content/en/blog/2024/otel-operator-q-and-a/index.md

```diff
@@ -0,0 +1,345 @@
+and
+[OpenTelemetry Collector Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib).
+
+If you want to use your own specific Collector components, you can build your
+own distribution using the
+[OpenTelemetry Collector Builder](/docs/collector/custom-collector/) (OCB), and
+include only the components that you need.
+
+Either way, the OpenTelemetryCollector CR allows you to override the default
+Collector image with one that better suits your needs by `adding spec.image` to
+your `OpenTelemetryCollector` YAML. In addition, you can also specify the number
+of Collector replicas by adding `spec.replicas`. This is totally independent of
+whether or not you override the Collector image.
+
+Your code would look something like this:
+
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: mynamespace
+spec:
+  mode: statefulset
+  image: <my_collector_image>
+  replicas: <number_of_replicas>
+```
+
+Where:
+
+- `<my_collector_image>` is the name of a valid Collector image from a container
+  repository
+- `<number_of_replicas>` is the number of pod instances for the underlying
+  OpenTelemetry Collector
+
+Keep in mind that if you're pulling a Collector image from a private container
+registry, you'll need to use
+[`imagePullSecrets`](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).
+Since private container registries require authentication, this will enable you
+to authenticate against that private registry. For more info on how to use
+`imagePullSecrets` for your Collector image, check out the instructions
+[here](https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#using-imagepullsecrets).
+
+For more info, check out the
+[OpenTelemetryCollector CR API docs](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollector).
+
+### Q5: Does the Target Allocator work for all deployment types?
+
+No. The Target Allocator only works for
+[StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/),
+and
+[DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/)
+([newly-introduced](https://github.com/open-telemetry/opentelemetry-operator/pull/2430#discussion_r1420495631)).
+More info
+[here](https://github.com/open-telemetry/opentelemetry-operator/blob/aed905c2c3c0aa3fb608a79c2e4d0e7b73dff980/apis/v1beta1/collector_webhook.go#L328).
+
+### Q6: If I’m using Operator’s Target Allocator for Prometheus service discovery, do I need `PodMonitor` and `ServiceMonitor` CRs installed in my Kubernetes cluster?
+
+Yes, you do. These CRs are bundled with the
+[Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator);
+however, they can be installed standalone, which means that you don’t need to
+install the Prometheus Operator just to use these two CRs with the Target
+Allocator.
```

Comment:
This is actually not correct. You need the CRs if you want to use them (duh), but you can also write raw Prometheus scrape configs using service discovery, and the target allocator will handle them correctly. You need to do this if you want to use the kinds of service discovery PodMonitors and ServiceMonitors don't support - for example if you want to scrape cAdvisor or kubelet metrics.

The answer to this question is already a bit complicated though, so maybe we should add a separate question for it? Unfortunately, "Prometheus service discovery" can have different meaning depending on context, and some readers may associate it purely with the CRs.

Reply:
@swiatekm-sumo I re-worded the question to be more specific to what I was trying to address: 

> If I enable `prometheusCR` in the Target Allocator, do I need the `PodMonitor` and `ServiceMonitor` CRs installed in my Kubernetes cluster?

What do you think?